### DISCUSSION TITLE
RandomInterface share the generated random number across different kernels

### DISCUSSION THREAD

**Discussion by KhaledNabilSharafeldin** [idaholab/moose#21496] [Q&A General] [answered]
Hello,
I am using the RandomInterface to generate a random number inside a kernel. Is there a way to share this exact number for different kernels on the same node?

Thanks in advance,
Khaled

**Comment by GiudGiud** [comment, marked as answer]
Hello

Looking at the doxygen, I dont think so. Every time the getRandomReal() is called for that node, it gets a new number.
https://mooseframework.inl.gov/docs/doxygen/moose/classRandomInterface.html

You could save this real in a material property or an auxvariable if you want to share it across multiple objects on the same elements/qps

Guillaume

**Reply by KhaledNabilSharafeldin** [reply]
Thank you @GiudGiud! I ended up setting up a user object to do just that, it executes in timestep_begin and has a getter function that passes the result without producing new random numbers. I can share the UO for reference here, if it's needed.

But I have another question because the documentation on RandomInterface might be throwing me off a bit. [it](https://mooseframework.inl.gov/source/interfaces/RandomInterface.html) is saying that 
> ...you don't want to be producing new random numbers with each linear iteration as this can impact the convergence of the solver. You can avoid this problem If you "reset" the random number generators every "linear" iteration...

so that means if I set the `setRandomResetFrequency(EXEC_LINEAR)` it would keep generating the same number over and over during these linear itterations? what about the next time step? would it generate the same number again? does the seed change after it is being set initially?

The reason I am asking this is because I keep getting the same pattern in my run, I will need to look more into it, but even if I use `time(NULL)` to provide a unique seed other than the default it seems to be almost identical result.

Thanks again for the reply!
Best,
Khaled

**Reply by GiudGiud** [reply]
Ah this is a good point. I think it ll indeed reset all the time, between linear iterations.
The random numbers only depend on the location, the seed and the number of numbers previously generated it seems.

I think you should modify the seed of the randomData object at each time step if you want to change the pattern between time steps. 

Note that you are getting the same seed between runs on purpose. This really helps debugging to keep the same seed
